numpy/core/arrayprint.py

```diff
@@ -1209,6 +1211,31 @@ def array_repr(arr, max_line_width=None, precision=None, suppress_small=None):
             lf = '\n'+' '*len(class_name + "(")
         return "%s(%s,%sdtype=%s)" % (class_name, lst, lf, typename)
 
+
+    if issubclass(arr.dtype.type, flexible):
```

Comment:
This entire `if` can be placed before the `if _format_options['legacy']` to save on code duplication. That way, the legacy case becomes simply:

```
if _format_options['legacy'] == '1.13':
    if issubclass(arr.dtype.type, flexible):
        lf =  '\n'+' '*len(class_name + "(")
    else:
        lf = ''
    return "%s(%s,%sdtype=%s)" % (class_name, lst, lf, typename)
```